File: ReadyTasks/CSharp/ProjectEuler/Task_29/Task_29/Program.cs
using System;
using System.Numerics;
using System.Collections.Generic;

namespace Task_29
{
    class Program
    {
        static int GetDigitSum(BigInteger numb)
        {
            int result = 0;
            while (numb > 0)
            {
                result += (int)(numb % 10);
                numb /= 10;
            }
            return result;
        }

        static int GetResult(int minA, int maxA, int minB, int maxB)
        {
            HashSet<BigInteger> hashset = new HashSet<BigInteger>();
            for (int a = minA; a <= maxA; a++)
            {
                for (int b = minB; b <= maxB; b++)
                {
                    hashset.Add(BigInteger.Pow(a, b));
                }
            }
            return hashset.Count;
        }

        static void Main(string[] args)
        {
            Console.WriteLine(GetResult(2, 100, 2, 100));
        }
    }
}

File: ReadyTasks/FSharp/ProjectEuler/Task_29/Task_29/Program.fs
open System

let getResult minA maxA minB maxB =
    [minA..maxA] |> List.allPairs [minB..maxB] |> List.map (fun (a: int,b) -> bigint.Pow(bigint(a), b)) |>
    List.distinct |> List.length

[<EntryPoint>]
let main argv =
    printfn "%A" (getResult 2 100 2 100)
    0 // return an integer exit code